plugins/cake/cake_

```diff
@@ -0,0 +1,202 @@
+			if [ "$qdisc" == "cake" ]; then
+				cake_ifs+=("$if")
+			fi
+		done
+		echo "$cake_ifs"
+	fi
+	exit 0
+	;;
+	config)
+	
+	if [ $has_if_in -eq 0 ]; then
+		echo "graph_title $DEVICE ($DEVICE_IN) cake queuing"
+	else
+		echo "graph_title $DEVICE cake queuing"
+	fi
+	echo 'graph_args --base 1000'
+	if [ $has_if_in -eq 0 ]; then
+		echo 'graph_vlabel pps in (-) / out (+)'
+	else
+		echo 'graph_vlabel packets per second'
+	fi
+	echo 'graph_category network'
+	if [ $has_if_in -eq 0 ]; then
+		echo "graph_info This graph shows the general packet handling status of egress+ingress traffic of the $DEVICE ($DEVICE_IN) network interface."
+	else
+		echo "graph_info This graph shows the general packet handling status of egress traffic of the $DEVICE network interface."
+	fi
+	
+	if [ $has_if_in -eq 0 ]; then
+		echo "backlog_in.label backlog (absolute)"
+		echo "backlog_in.draw LINE2"
+		echo "backlog_in.min 0"
+		echo "backlog_in.info amount of packets currently buffered"
+		echo "backlog_in.graph no"
+	fi
+	echo "backlog.label backlog (absolute)"
+	echo "backlog.draw LINE2"
+	echo "backlog.min 0"
+	echo "backlog.info amount of packets currently buffered"
+	[ $has_if_in -eq 0 ] && echo "backlog.negative backlog_in"
+	
+	if [ $has_if_in -eq 0 ]; then
+		echo "dropped_in.label dropped"
+		echo "dropped_in.draw AREA"
+		echo "dropped_in.type DERIVE"
+		echo "dropped_in.min 0"
+		echo "dropped_in.info dropped packets in queue"
+		echo "dropped_in.graph no"
+	fi
+	echo "dropped.label dropped"
+	echo "dropped.draw AREA"
+	echo "dropped.type DERIVE"
+	echo "dropped.min 0"
+	echo "dropped.info dropped packets in queue"
+	[ $has_if_in -eq 0 ] && echo "dropped.negative dropped_in"
+	
+	if [ $has_if_in -eq 0 ]; then
+		echo "overlimits_in.label overlimits"
+		echo "overlimits_in.draw STACK"
+		echo "overlimits_in.type DERIVE"
+		echo "overlimits_in.min 0"
+		echo "overlimits_in.info packets exeeded a limit"
+		echo "overlimits_in.graph no"
+	fi
+	echo "overlimits.label overlimits"
+	echo "overlimits.draw STACK"
+	echo "overlimits.type DERIVE"
+	echo "overlimits.min 0"
+	echo "overlimits.info packets exeeded a limit"
+	[ $has_if_in -eq 0 ] && echo "overlimits.negative overlimits_in"
+	
+	if [ $has_if_in -eq 0 ]; then
+		echo "requeues_in.label requeues"
+		echo "requeues_in.draw STACK"
+		echo "requeues_in.type DERIVE"
+		echo "requeues_in.min 0"
+		echo "requeues_in.info packets requeued in queue"
+		echo "requeues_in.graph no"
+	fi
+	echo "requeues.label requeues"
+	echo "requeues.draw STACK"
+	echo "requeues.type DERIVE"
+	echo "requeues.min 0"
+	echo "requeues.info packets requeued in queue"
+	[ $has_if_in -eq 0 ] && echo "requeues.negative requeues_in"
+	
+	exit 0
+	;;
+esac
+
+tc_cake_sent "$DEVICE" | awk '{
+	print "dropped.value " $7
+	print "overlimits.value " $9
+	print "requeues.value " $11
+}'
+
+tc_cake_backlog "$DEVICE" | awk '{
+	print "backlog.value " $3
+}'
+
+if [ $has_if_in -eq 1 ]; then
```

Comment:
The logic of conditional should be clearer for the next reader, if you invert the expression and place the two blocks that are only relevant in case of an existing interface into the `then` branch.

Reply:
Why do you would want to invert this?

Reply:
You are using an approach like `early return` here: the conditional secretly applies to all following lines.
Thus a casual contributor, who would add another graph at the end of the plugin would be surprised, that it is not included under some circumstances. Keeping the scope of a conditional small and obvious is usually a good thing :)

Reply:
It's quite early in the morning, but I don't really get what you want. 🤔

There are currently very small blocks of output conditional, not big chunks. How do you want me to change this?

Reply:
Thank you!
Next time I will try to be more precise ...